org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -207,7 +216,7 @@ private IMarker getExistingMarkerFor(IDocument document, Diagnostic diagnostic,
 				if (LSPEclipseUtils.toOffset(diagnostic.getRange().getStart(), document) == MarkerUtilities.getCharStart(marker)
 						&& LSPEclipseUtils.toOffset(diagnostic.getRange().getEnd(), document) == MarkerUtilities.getCharEnd(marker)
 						&& Objects.equals(marker.getAttribute(IMarker.MESSAGE), diagnostic.getMessage())
-						&& Objects.equals(marker.getAttribute(LANGUAGE_SERVER_ID), this.languageServerId)) {
+						&& Objects.equals(marker.getAttribute(LANGUAGE_SERVER_ID), languageServerDefinition)) {
```

Comment:
I believe marker do require Serializable objects of something of that sort as they are persisted in the workspace. If so, we'll need to stick to id.

Reply:
Okay, then back to the roots. If it passes the tests I will merge it. We can always improve later if wanted